ui/src/main/webapp/home/actions.js

```diff
@@ -0,0 +1,37 @@
+import { poll } from 'redux-polling'
+import { get, post } from 'redux-fetch'
+
+// actions
+
+export const setConfigs = (id, value) => ({type: 'HOME/SET_CONFIGS', id, value})
+export const beginDelete = (id, servicePid) => ({type: 'HOME/BEGIN_SUBMITTING_CONFIGS', id, servicePid})
+
+// async actions
+
+export const retrieve = (id) => async (dispatch) => {
+  const res = await dispatch(get('/admin/beta/config/configurations/' + id))
+  const json = await res.json()
+
+  if (res.status === 200) {
+    dispatch(setConfigs(id, json))
+  }
+}
+
+export const refresh = poll('home', () => (dispatch) =>
+  Promise.all([
+    dispatch(retrieve('sources')),
+    dispatch(retrieve('ldap'))
+  ])
+)
+
+export const deleteConfig = ({ configurationType, factoryPid, servicePid, id }) => async (dispatch) => {
+  dispatch(beginDelete(id, servicePid))
+  const url = '/admin/beta/config/persist/' + configurationType + '/delete'
+  const body = JSON.stringify({configurationType, factoryPid, servicePid})
+
+  const res = await dispatch(post(url, {body}))
+
+  if (res.status === 200) {
```

Comment:
Ya that's something that should definitely be done. It's a bit outside the ticket scope, but I'll go ahead and create an error message reporting component for the home page config tiles.